File: PersonnelManagement/PersonnelManagement/Models/Position.cs
using System;
using System.Collections.Generic;

namespace PersonnelManagement.Models
{
    public partial class Position
    {
        public int Id { get; set; }
        public int? Cap { get; set; }
        public DateTime? Dateactive { get; set; }
        public DateTime? Dateinactive { get; set; }
        public int Sourceoffinancing { get; set; }
        public int Positiontype { get; set; }
        public string Notice { get; set; }
        public int Positioncategory { get; set; }
        public sbyte Replacedbycivil { get; set; }
        public int Replacedbycivilpositioncategory { get; set; }
        public int Replacedbycivilpositiontype { get; set; }
        public sbyte Altrank { get; set; }
        public int Origin { get; set; }
        public sbyte Decertificate { get; set; }
        public DateTime? Decertificatedate { get; set; }
        public int Civilranklow { get; set; }
        public int Civilrankhigh { get; set; }
        public sbyte Replacedbycivildatelimit { get; set; }
        public DateTime? Replacedbycivildate { get; set; }
        public int Structure { get; set; }
        public sbyte Civildecree { get; set; }
        public string Civildecreenumber { get; set; }
        public DateTime? Civildecreedate { get; set; }
        public sbyte Curator { get; set; }
        public sbyte Head { get; set; }
        public string Curatorlist { get; set; }
        public int Headid { get; set; }
        public sbyte Opchs { get; set; }
        public sbyte Part { get; set; }
        public double Partval { get; set; }
        public int Subject1 { get; set; }
        public int Subject2 { get; set; }
        public int Subject3 { get; set; }
        public int Subject4 { get; set; }
        public int Subject5 { get; set; }
        public int Subject6 { get; set; }
        public int Subject7 { get; set; }
        public int Subject8 { get; set; }
        public int Subject9 { get; set; }
        public int Subject10 { get; set; }
        public int Subject11 { get; set; }
        public int Subject12 { get; set; }
        public int Subject13 { get; set; }
        public int Subject14 { get; set; }
        public int Subject15 { get; set; }
        public int Subject16 { get; set; }
        public int Subject17 { get; set; }
        public int Subject18 { get; set; }
        public int Subject19 { get; set; }
        public int Subject20 { get; set; }
        public string Name1 { get; set; }
        public string Name2 { get; set; }
        public string Name3 { get; set; }
        public string Name4 { get; set; }
        public string Name5 { get; set; }
        public string Name6 { get; set; }
        public uint Subjectindex { get; set; }
    }
}

File: PersonnelManagement/PersonnelManagement/Front/classes/position.ts
export default class Position {
    id: number;
    name: string;
    department: number;
    photo: Blob;
    cap: number;
    sourceoffinancing: number;
    positiontype: number;
    positioncategory: number;
    notice: string;
    replacedbycivil: number;
    replacedbycivilpositioncategory: number; // can be 687 means 6, 8, 7
    replacedbycivilpositiontype: number;
    mrd: string; // Ids of mrd listed as comma separated values "3,11,8"
    decertificate: number;
    decertificatedate: Date;
    civilranklow: number;
    civilrankhigh: number;
    replacedbycivildatelimit: number;
    replacedbycivildate: Date;
    civildecree: number;
    civildecreenumber: string;
    civildecreedate: Date;

    structure: number;
    parenttype: number;

    curator: number;
    curatorlist: string;
    head: number;
    headid: number;
    opchs: number;
    part: number;
    partval: number;

    subject1: number;
    subject2: number;
    subject3: number;
    subject4: number;
    subject5: number;
    subject6: number;
    subject7: number;
    subject8: number;
    subject9: number;
    subject10: number;
    subject11: number;
    subject12: number;
    subject13: number;
    subject14: number;
    subject15: number;
    subject16: number;
    subject17: number;
    subject18: number;
    subject19: number;
    subject20: number;
    name1: string;
    name2: string;
    name3: string;
    name4: string;
    name5: string;
    name6: string;
}